include/marl/memory.h

```diff
@@ -203,7 +226,7 @@ std::shared_ptr<T> Allocator::make_shared(ARGS&&... args) {
 
   auto alloc = allocate(request);
   new (alloc.ptr) T(std::forward<ARGS>(args)...);
-  return std::shared_ptr<T>(reinterpret_cast<T*>(alloc.ptr), Deleter{this});
+  return std::shared_ptr<T>(reinterpret_cast<T*>(alloc.ptr), Deleter{this, 1});
```

Comment:
I had not noticed this before but:
std::shared_ptr<> typically allocates more memory internally, so you may end up with a heap allocation that is not caught by your allocator. Not sure how much this matters, but it's something I noticed.

Reply:
Great catch. Yes - I hadn't thought about this either, but you're right. Unfortunately there's no `make_shared` that takes an allocator. Grr - looks like I'm going to have to roll my own ref-counted types.

Mentioning https://github.com/google/marl/issues/131 so this isn't forgotten.

Reply:
@AWoloszyn just pointed me at [`std::allocate_shared`](https://en.cppreference.com/w/cpp/memory/shared_ptr/allocate_shared) which looks like it might do exactly what we need. Thanks Andrew!